Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43890&attempt_id=7b2b1cb7-0d1a-4f34-8982-fef25db914ed)

Attempted: tests.cross.mingw32, pkgsCross.mingw32.openssl, pkgsCross.mingwW64.openssl, pkgsCross.mingw32.libxslt, pkgsCross.mingwW64.libxslt, pkgsCross.mingw32.ncurses, pkgsCross.mingwW64.ncurses, pkgsCross.mingw32.gettext, pkgsCross.mingwW64.gettext

The following builds were skipped because they don't evaluate on aarch64-linux: tests.cross.mingwW64

<details><summary>Partial log (click to expand)</summary><p>

```
make[4]: Leaving directory '/build/gettext-0.19.8.1/gettext-tools/gnulib-lib'
make[3]: *** [Makefile:2024: all-recursive] Error 1
make[3]: Leaving directory '/build/gettext-0.19.8.1/gettext-tools'
make[2]: *** [Makefile:1890: all] Error 2
make[2]: Leaving directory '/build/gettext-0.19.8.1/gettext-tools'
make[1]: *** [Makefile:413: all-recursive] Error 1
make[1]: Leaving directory '/build/gettext-0.19.8.1'
make: *** [Makefile:369: all] Error 2
builder for '/nix/store/vp9p01dgd60w8bqrfrq558ml0pqi2m1i-gettext-0.19.8.1-x86_64-pc-mingw32.drv' failed with exit code 2
error: build of '/nix/store/9a0islhwak25r9kz8hby63sx0rcy47ma-test-hello-2.10-i686-pc-mingw32.drv', '/nix/store/asc1a11vyjjq68n61sp9d4qd2vgq8b7s-test-file-5.33-winlinks-i686-pc-mingw32.drv', '/nix/store/vp9p01dgd60w8bqrfrq558ml0pqi2m1i-gettext-0.19.8.1-x86_64-pc-mingw32.drv' failed
```
</p></details>

--- a/pkgs/development/libraries/ncurses/default.nix
+++ b/pkgs/development/libraries/ncurses/default.nix
@@ -34,7 +34,11 @@ stdenv.mkDerivation rec {
   ] ++ lib.optional unicode "--enable-widec"
     ++ lib.optional enableStatic "--enable-static"
     ++ lib.optional (!withCxx) "--without-cxx"
-    ++ lib.optional (abiVersion == "5") "--with-abi-version=5";
+    ++ lib.optional (abiVersion == "5") "--with-abi-version=5"
+    ++ lib.optionals hostPlatform.isWindows [
+      "--enable-sp-funcs"
+      "--enable-term-driver"
+    ];
 
   # Only the C compiler, and explicitly not C++ compiler needs this flag on solaris:
   CFLAGS = lib.optionalString stdenv.isSunOS "-D_XOPEN_SOURCE_EXTENDED";

--- a/pkgs/development/libraries/openssl/default.nix
+++ b/pkgs/development/libraries/openssl/default.nix
@@ -45,7 +45,9 @@ let
         if hostPlatform == buildPlatform
           then "./config"
         else if hostPlatform.isMinGW
-          then "./Configure mingw${toString hostPlatform.parsed.cpu.bits}"
+          then "./Configure mingw${optionalString
+                                     (hostPlatform.parsed.cpu.bits != 32)
+                                     (toString hostPlatform.parsed.cpu.bits)}"
         else if hostPlatform.isLinux
           then "./Configure linux-generic${toString hostPlatform.parsed.cpu.bits}"
         else if hostPlatform.isiOS

--- a/pkgs/tools/misc/file/default.nix
+++ b/pkgs/tools/misc/file/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, file, zlib }:
+{ stdenv, fetchurl, file, zlib, libgnurx }:
 
 stdenv.mkDerivation rec {
   name = "file-${version}";
@@ -13,10 +13,13 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = stdenv.lib.optional (stdenv.hostPlatform != stdenv.buildPlatform) file;
-  buildInputs = [ zlib ];
+  buildInputs = [ zlib ]
+              ++ stdenv.lib.optional stdenv.hostPlatform.isWindows libgnurx;
 
   doCheck = true;
 
+  makeFlags = if stdenv.hostPlatform.isWindows then "FILE_COMPILE=file"
+              else null;
 
   meta = with stdenv.lib; {
     homepage = http://darwinsys.com/file;

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -2387,7 +2387,9 @@ with pkgs;
 
   figlet = callPackage ../tools/misc/figlet { };
 
-  file = callPackage ../tools/misc/file { };
+  file = callPackage ../tools/misc/file {
+    inherit (windows) libgnurx;
+  };
 
   filegive = callPackage ../tools/networking/filegive { };
 

